python/ray/data/impl/arrow_block.py

```diff
@@ -153,6 +173,85 @@ def _sample(self, n_samples: int, key: "SortKeyT") -> "pyarrow.Table":
         indices = random.sample(range(self._table.num_rows), n_samples)
         return self._table.select([k[0] for k in key]).take(indices)
 
+    def count(self, on: KeyFn) -> Optional[U]:
```

Comment:
Add a comment saying it counts non-null rows?

Reply:
That's mentioned in the docstring, are you wanting more information there?

Reply:
oops, I missed that.